client/reducers/cycles.js

```diff
@@ -50,6 +51,31 @@ export default (state = initialState, action) => {
                 }
             });
         }
+        case SAVE_RESULT: {
+            const result = action.payload;
+
+            const tests =
+                state.runsForCycle[result.cycle_id][result.run_id].tests;
```

Comment:
Is it guaranteed that there will always be `result.cycle_id` and `result.run_id`? Similarly, is it safe to get of `state.runsForCycle[...][...].tests;`?

Reply:
yes all of that data is definitely there if you are in the middle of a test run.